feat: use the new Workers Static Assets feature from Cloudflare (#13072)
sveltejs/kit: .changeset/smart-owls-trade.md

---
'@sveltejs/adapter-cloudflare-workers': major
---

feat: use the new Workers Static Assets feature from Cloudflare

> This changes the adapter to stop using the old Workers Sites (kv-asset-handler) approach.
> Instead, making use of the new Workers Static Assets feature, which is embedded into Cloudflare natively.

Reviewer: ```suggestion
This changes the adapter to stop using the old Workers Sites (kv-asset-handler) approach in favor of the new Workers Static Assets feature, which is embedded into Cloudflare natively.
```

> Also this change removes the extra esbuild step that was being run inside the adapter, relying upon Wrangler to do the bundling.

Reviewer: ```suggestion
Also, this change removes the extra esbuild step that was being run inside the adapter, instead relying upon Wrangler to do the bundling.
```

> The extra esbuild step required a hardcoded list of Node.js compatible modules.
> This is no longer needed since Wrangler now manages all of that.

Reviewer: I think we can probably drop this to keep things simpler
```suggestion
```

## Breaking changes and migration

- This version of the adapter requires Wrangler version 3.87.0 or later.

  Run `npm add -D wrangler@latest` (or similar) in your project to update Wrangler.
- The user's Wrangler configuration (`wrangler.toml`) must be migrated from using Workers Sites to using Workers Assets.

>   Previously a user's `wrangler.toml` might look like:

Reviewer: is there any migration guide on the cloudflare site we could point to instead of including the details here? most of our changelog entries are fairly minimal

  ```toml
  name = "<your-site-name>"
  account_id = "<your-account-id>"
  compatibility_date = "2021-11-12"
  main = "./.cloudflare/worker.js"

  # Workers Sites configuration
  site.bucket = "./.cloudflare/public"
  ```

  Change it to to look like:

  ```toml
  name = "<your-site-name>"
  account_id = "<your-account-id>"

>   compatibility_date = "2021-11-12"`

Reviewer 2: ```suggestion
  compatibility_date = "2021-11-12"
```

  main = ".svelte-kit/cloudflare/server/index.js"

  # Workers Assets configuration
  assets = { directory = ".svelte-kit/cloudflare/client" }
  ```

- Workers Assets defaults to serving assets directly for a matching request, rather than routing it through the Worker code.
  

>   The previous adapter would add custom headers to assets responses (such as `cache-control`, `content-type`, and `x-robots-tag`. Such direct asset responses no longer contain these headers - but the will include eTag headers that have proven (in Pages) to be an effective caching strategy for assets.

Reviewer: ```suggestion
  The previous adapter would add custom headers to assets responses (such as `cache-control`, `content-type`, and `x-robots-tag`. Such direct asset responses no longer contain these headers — but they will include eTag headers that have proven (in Pages) to be an effective caching strategy for assets.
```
